src/StreamJsonRpc/Reflection/RpcTargetInfo.cs

```diff
@@ -371,7 +365,18 @@ private static IReadOnlyDictionary<string, List<MethodSignature>> GetRequestMeth
             }
         }
 
-        void ActOn([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods | DynamicallyAccessedMemberTypes.NonPublicMethods)] TypeInfo t)
+        [UnconditionalSuppressMessage("Trimming", "IL2062:UnrecognizedReflectionPattern", Justification = "exposedMembersOnType is annotated as preserve All members, so any Types returned from GetInterfaces should be preserved as well. See https://github.com/dotnet/linker/issues/1731.")]
+        void ActOnInterfaces([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] Type exposedMembersOnType)
+        {
+            foreach (Type iface in exposedMembersOnType.GetInterfaces())
+            {
+#pragma warning disable IL2072 // The Roslyn analyzer warns a different error code than the ilc compiler.
+                ActOn(iface.GetTypeInfo());
+#pragma warning restore IL2072
+            }
+        }
```

Comment:
Nit: if you rewrite this to use `for (int i = 0; ...)` (to work around a Roslyn trim analyzer limitation around `foreach`), the suppression could be under a NET9_OR_BELOW ifdef and eventually becomes dead code that could be found and cleaned up at some point.

(I really hate suppressions.)

Reply:
Nice! I didn't realize that was the problem. Fixed.